test/api/test_workflows.py

```diff
@@ -2135,6 +2138,54 @@ def test_run_with_validated_parameter_connection_valid(self):
             jobs = self._history_jobs(history_id)
             assert len(jobs) == 1
 
+    def test_run_with_int_parameter(self):
+        with self.dataset_populator.test_history() as history_id:
+            failed = False
+            try:
+                self._run_jobs(WORKFLOW_PARAMETER_INPUT_INTEGER_REQUIRED, test_data="""
+data_input:
+  value: 1.bed
+  type: File
+""", history_id=history_id, wait=True, assert_ok=True)
+            except AssertionError as e:
+                assert '(int_input) has no input' in str(e)
+                failed = True
+            assert failed
+            self._run_jobs(WORKFLOW_PARAMETER_INPUT_INTEGER_REQUIRED, test_data="""
+data_input:
+  value: 1.bed
+  type: File
+int_input:
+  value: 1
+  type: raw
+""", history_id=history_id, wait=True, assert_ok=True)
+            self.dataset_populator.wait_for_history(history_id, assert_ok=True)
+            content = self.dataset_populator.get_history_dataset_content(history_id)
+            assert len(content.split("\n")) == 2, content
+
+            failed = False
+            try:
+                self._run_jobs(WORKFLOW_PARAMETER_INPUT_INTEGER_OPTIONAL, test_data="""
+data_input:
+  value: 1.bed
+  type: File
+""", history_id=history_id, wait=True, assert_ok=True)
+            except AssertionError as e:
+                print(e)
+                failed = True
+            assert not failed
+
+    def test_run_with_validated_parameter_connection_default_values(self):
+        with self.dataset_populator.test_history() as history_id:
+            self._run_jobs(WORKFLOW_PARAMETER_INPUT_INTEGER_DEFAULT, test_data="""
+data_input:
+  value: 1.bed
+  type: File
+""", history_id=history_id, wait=True, assert_ok=True)
+            self.dataset_populator.wait_for_history(history_id, assert_ok=True)
+            content = self.dataset_populator.get_history_dataset_content(history_id)
+            assert len(content.split("\n")) == 4, content
```

Comment:
```suggestion
            assert len(content.splitlines()) == 3, content
```
that feels more intuitive